src/daft-scan/src/lib.rs

```diff
@@ -124,6 +125,7 @@ pub enum DataFileSource {
         metadata: Option<TableMetadata>,
         partition_spec: Option<PartitionSpec>,
         statistics: Option<TableStatistics>,
+        parquet_metadata: Option<Arc<FileMetaData>>,
```

Comment:
I believe that `CatalogDataFile` would also benefit from `parquet_metadata` caching